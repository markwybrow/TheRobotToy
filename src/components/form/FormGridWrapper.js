import React, { useState, useEffect, useContext } from 'react';
import { RobotContext } from '../../context/RobotContext';
import { Board } from '../common/playingboard';
import PropTypes from 'prop-types';

const FormGridWrapper = () => {
  const { state, dispatch } = useContext(RobotContext);
  const { gridDimensions, gridPosition } = state;
  //	const [gridSettings, setGrid] = useState(gridDimensions);
  const [gridPositionSettings, setGridPosition] = useState(gridPosition);

  let BoardConfig = { ...gridDimensions, ...gridPositionSettings };

  useEffect(
    () => {
      BoardConfig = {
        ...gridDimensions,
        ...gridPositionSettings,
      };
      gridPositionHandler();
    },
    [gridPositionSettings, gridDimensions],
  );

  const submitFormHandler = (event) => {
    event.preventDefault();
  };

  const gridDimensionsHandler = (event) => {
    dispatch({
      type: 'changeGridDimensions',
      newGridDimensions: {
        ...gridDimensions,
        [event.target.name]: parseInt(event.target.value),
      },
    });
  };

  const gridPositionHandler = (event) => {
    event &&
      setGridPosition({
        ...gridPositionSettings,
        [event.target.name]: event.target.value,
      });

    dispatch({
      type: 'changeGridPosition',
      newGridPosition: { ...gridPositionSettings },
    });
  };
  return (
    <div>
      <form onSubmit={submitFormHandler}>
        <div className="">
          <div className="container column">
            <h5>Grid Size:</h5>

            <div className="form-row">
              <label className="label">Grid X:</label>

              <input
                name="gridX"
                value={gridDimensions.gridX}
                placeholder="Horizontal"
                type="number"
                onChange={gridDimensionsHandler}
              />
            </div>

            <div className="form-row">
              <label className="label">Grid Y:</label>

              <input
                name="gridY"
                value={gridDimensions.gridY}
                placeholder="Vertical"
                type="number"
                onChange={gridDimensionsHandler}
              />
            </div>
          </div>

          <div className="container column">
            <h5 className="">Start Position:</h5>
            <div className="form-row">
              <label className="label">X Value:</label>
              <input name="posX" value={gridPosition.posX} type="number" onChange={gridPositionHandler} />
            </div>

            <div className="form-row">
              <label className="label">Y Value:</label>

              <input name="posY" value={gridPosition.posY} type="number" onChange={gridPositionHandler} />
            </div>

            <div className="form-row">
              <label className="label">Direction:</label>

              <select name="direction" value={gridPosition.direction} onChange={gridPositionHandler}>
                <option value="N">NORTH</option>
                <option value="S">SOUTH</option>
                <option value="E">EAST</option>
                <option value="W">WEST</option>
              </select>
            </div>
          </div>
          <div className="container">
            <h5 className="">Platform:</h5>
            <Board {...BoardConfig} />
          </div>
        </div>
      </form>
    </div>
  );
};

FormGridWrapper.propTypes = {
  placeholder: PropTypes.string,
};

export default FormGridWrapper;
