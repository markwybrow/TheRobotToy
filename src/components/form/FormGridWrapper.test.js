import React from 'react';
import { shallow } from 'enzyme';

import FormGridWrapper from './FormGridWrapper';

describe('FormGridWrapper', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<FormGridWrapper debug />);

    expect(component).toMatchSnapshot();
  });
});
