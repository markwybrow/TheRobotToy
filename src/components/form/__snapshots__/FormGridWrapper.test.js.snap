// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should render correctly in "debug" mode 1`] = `
<div
  className="control"
>
  <input
    className="input is-small"
    onChange={[Function]}
    placeholder="text"
    type="text"
  />
</div>
`;
